src/core/trulens/core/utils/python.py

```diff
@@ -535,37 +543,43 @@ def merge_stacks(
 
     ret = []
 
-    while len(s1) > 1:
-        f = s1[0]
-        s1 = s1[1:]
-
+    for f in s1:
         ret.append(f)
         try:
             s2i = s2.index(f)
             for _ in range(s2i):
                 ret.append(s2[0])
                 s2 = s2[1:]
-
         except Exception:
             pass
 
     return ret
 
 
-def stack_with_tasks() -> Sequence[FrameType]:
+def stack_generator(offset: int = 0) -> Iterable[FrameType]:
+    frame = inspect.currentframe()
+    for _ in range(offset):
+        assert frame is not None
```

Comment:
Do we have to use assert here? or can we just raise exception with error message